packages/core/code-evaluation/rebuildGeometryCli.js

```diff
@@ -10,34 +10,27 @@ const { registerAllExtensions } = require('../io/registerExtensions')
 
 const rebuildSolids = (data) => {
   const defaults = { vtreeMode: false, serialize: false }
-  const { mainPath, vtreeMode, parameterValues, inputIsDirectory } = Object.assign({}, defaults, data)
+  let { mainPath, vtreeMode, parameterValues, inputIsDirectory } = Object.assign({}, defaults, data)
   const apiMainPath = vtreeMode ? '../code-loading/vtreeApi' : '@jscad/modeling'
   // we need to update the source for our module
   let requireFn = require
-  if (!inputIsDirectory) {
+
+  // source came from conversion, i.e. not from file system
+  if (data.source) {
     let filesAndFolders = [
       {
-        ext: path.extname(mainPath).substring(1),
-        name: path.basename(mainPath),
-        fullPath: mainPath,
+        ext: 'js',
+        fullPath: './index.js',
+        name: 'index.js',
         source: data.source
       }
     ]
-    filesAndFolders = transformSources({ apiMainPath }, filesAndFolders)
-    // console.log('filesAndFolders', filesAndFolders)
+    requireFn = makeWebRequire(filesAndFolders, { apiMainPath })
 
-    const fakeFs = makeFakeFs(filesAndFolders)
-    requireFn = makeWebRequire(filesAndFolders, { apiMainPath, fakeFs })// hasRequire() ? require : makeWebRequire(filesAndFolders, { apiMainPath })
-    // register all extension formats
-    registerAllExtensions(fakeFs, requireFn)
+    mainPath = './index.js' // and use the alias as the entry point
   }
 
-  // console.log('transformed sources', filesAndFolders)
-  // now check if we need fake require or not
-  // FIXME: we need to come up with a way to intercept node 'require' calls to be able to apply transformSources on the fly
```

Comment:
Unless this is fixed, please leave as is :)
Removing console.log() debug statements is good, but TODO/FIXMEs are there for a reason

Reply:
> FIXME: we need to come up with a way to intercept node 'require' calls to be able to apply transformSources on the fly

I think is is fixed. The call to registerAllExtensions() makes this possible. And a very cool solution as well.